### DISCUSSION TITLE
Persistent NFS (SOLVED)

### DISCUSSION THREAD

**Discussion by knightflyer** [ltsp/ltsp#463] [Q&A]
Being rather new to ltsp but not LINUX.  I have created an LTSP server and did the chrootless image ( ltsp image /)
Everything works greate at first glance but the data is not being shared.
Our application resides on the server as /u.
I created a group <asset> and assigned my users to that group with rw.
Everyone can login via ipxe and cd to /u and run the application, the mysql is shared, but the text data is not being shared among the group.
I would like the /u on the server available as persistently shared to everyone in the group.

Our current production server serves the /u folder with NFS shares and allows remote mysql connnections.
All desktop pcs connect and share the /u application and data via NFS and the mysql database server.
This has worked quite well for at least a decade, but updating the client computers suck.

My goal is to simplify everything with LTSP, where did I go wrong?

Current OS is mint 20.1 xfce.

**Comment by alkisg** [comment]
As you have it now, the /u directory gets published via the read-only /srv/ltsp/images/x86_64.img image.
Wait what, read only?
Yes, any rootfs changes in ltsp clients only happen in RAM and are lost after reboot.

Put it in /srv/u instead, because the /srv directory by default doesn't get published to the image.
Then use the [FSTAB_x parameter of ltsp.conf](https://ltsp.org/man/ltsp.conf/) to mount /srv/u via NFS in rw mode.

**Reply by knightflyer** [reply]
can I create a link from /u /srv/u?
or do I need to place the data into /srv/u first? and then link to /u?

Would this allow a user on the server to access the common data?
thanks

**Comment by alkisg** [comment]
If you want to keep the location /u, just put an ADD_IMAGE_EXCLUDES="u/*" parameter in [ltsp.conf](https://ltsp.org/man/ltsp.conf/).
I only proposed /srv/u to make things simpler as /srv is the proper location for server data.

In any case, the main idea is that you still mount the folder to the clients using NFS rw mode.

**Reply by knightflyer** [reply]
Do I create a /etc/ltsp/ltsp.conf file and edit there bfore I re-image?

**Comment by alkisg** [comment]
Yes to both questions. For how to create it, read:
https://ltsp.org/man/ltsp.conf/#creation

**Reply by knightflyer** [reply]
while booting the client I see a couple of errors scrolling by. Where can I find the log on the client?

And should the ...
ADD_IMAGE_EXCLUDES="u/* 
be changed to  ...
ADD_IMAGE_EXCLUDES="/u/*

**Comment by knightflyer** [comment]
It seems the ADD_MAGES_EXLUDES="u/* in /etc/ltsp/ltsp.conf did not exclude the contents of u/* as desired
but when I put that entry in /usr/share/ltsp/server/image/image.excludes all worked as desired.
I could determine this by unmounting /u the nfs share as sudo and the read only /u persisted. 

Thank you alkisg in your help.

**Comment by alkisg** [comment]
Hi knightflyer, please note the three typos (I, S, "),
it's: `ADD_IMAGE_EXCLUDES="u/*"`
not: `ADD_MAGES_EXLUDES="u/*`

**Reply by knightflyer** [reply]
yep, those were my typos on the reply, but when I put the correct entry in /usr/share/ltsp/server/image/image.excludes all worked as desired.

I'm happy.  It's hard to believe the thin clients had enough RAM for the 3gig of crap stored in the /u directrory, squashfs must be efficient.
